pkg/provider/provider.go

```diff
@@ -30,10 +30,16 @@ func Provider() *schema.Provider {
 			},
 			"password": &schema.Schema{
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("SNOWFLAKE_PASSWORD", nil),
 				Sensitive:   true,
 			},
+			"browser_auth": &schema.Schema{
```

Comment:
Both `password` and `browser_auth` should have [ConflictsWith](https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/a19745031a0bd81ff66556212bfb3c7ebec730db/vendor/github.com/hashicorp/terraform/helper/schema/schema.go#L170) set to make it clear that you can't use both at the same time.